dotCMS/src/main/java/com/dotmarketing/portlets/templates/design/bean/ContainerUUID.java

```diff
@@ -23,11 +30,58 @@ public class ContainerUUID implements Serializable{
     private final String identifier;
     private String uuid;
 
+    /**x
+     * History of Changes: This section lists all the UUIDs that have been assigned to this Container. For example,
+     * if the history is ["3", "1", "4"], it means the Container was initially assigned UUID 3,
+     * then changed to UUID 1 after a layout modification, and is currently assigned UUID 4.
+     *
+     * If the layout changes but the Container is not moved, the same UUID will be repeated in the history.
+     * For instance, consider the following layout:
+     *
+     * Row 1: Container A, UUID 1, history: ["1"]
+     * Row 2: Container A, UUID 2, history: ["2"]
+     * If a new row is added after the first one, the UUIDs and histories will be as follows:
+     *
+     * Row 1: Container A, UUID 1, history: ["1", "1"]
+     * Row 2: Container A, UUID 2, history: ["2"]
+     * Row 3: Container A, UUID 3, history: ["2", "3"]
+     * The ["1", "1"] indicates that the layout changed, but this Container was not moved.
+     */
+    private List<String> historyUUIDs;
+
+    public ContainerUUID(final String containerIdOrPath,
+                         final String containerInstanceID) {
+
+        this.identifier = containerIdOrPath;
+        this.uuid = containerInstanceID == null ? UUID_DEFAULT_VALUE : containerInstanceID;
+        this.historyUUIDs = isNew(containerInstanceID) ? new ArrayList<>() : list(this.uuid);
+    }
+
     public ContainerUUID(final @JsonProperty("identifier") String containerIdOrPath,
-                         final @JsonProperty("uuid") String containerInstanceID) {
+                         final @JsonProperty("uuid") String containerInstanceID,
+                         final @JsonProperty("historyUUIDs") List<String> historyUUIDs) {
 
         this.identifier = containerIdOrPath;
         this.uuid = containerInstanceID == null ? UUID_DEFAULT_VALUE : containerInstanceID;
+
+        if (isNew(containerInstanceID)) {
+            this.historyUUIDs = new ArrayList<>();
+        } else {
+            this.historyUUIDs = UtilMethods.isSet(historyUUIDs) ? new ArrayList<>(historyUUIDs) : list(this.uuid);
+            ;
```

Comment:
Remove this colon.

Reply:
done https://github.com/dotCMS/core/pull/29752/commits/b16f6562e59d264efd3c344bd7164dec250da377